source/pick-deep.d.ts

```diff
@@ -0,0 +1,79 @@
+import type {BuildObject} from './internal';
+import type {Simplify} from './simplify.d';
+import type {UnionToIntersection} from './union-to-intersection.d';
+import type {UnknownRecord} from './unknown-record.d';
+
+type ToString<T> = T extends string | number ? `${T}` : '';
+
+/**
+Generate a union of all the possible paths in a deeply nested object.
+*/
+export type Paths<T extends UnknownRecord, RecordType extends UnknownRecord = Required<T>> =
```

Comment:
This should not be exported here. Instead, in a different PR, it could be moved to a separate file and become a public type: https://github.com/sindresorhus/type-fest/issues/213